src/discordtipbot/discordbot.cr

```diff
@@ -533,6 +535,46 @@ class DiscordBot
     end
   end
 
+  def grouplucky(msg : Discord::Message, cmd_string : String)
+    return reply(msg, "**ERROR**: This command doesn't work in DMs") if private?(msg)
+
+    cmd_usage = "#{@config.prefix}lucky [amount]"
+
+    # cmd[0]: command, cmd[1]: amount"
+    cmd = cmd_string.split(" ")
+
+    return reply(msg, cmd_usage) unless cmd.size > 1
+
+    amount = amount(msg, cmd[1])
+    return reply(msg, "**ERROR**: You have to specify an amount! #{cmd_usage}") unless amount
+
+    people = amount(msg, cmd[2]) # number of people to get lucky.
+
+    users = active_users(msg)
+
+    return reply(msg, "**ERROR**: There arent that many people!") unless (users = users.to_a).size > people
+
+    return reply(msg, "**ERROR**: You have to lucky rain at least #{@config.min_rain_total} #{@config.coinname_short}") unless amount >= @config.min_tip
+
+    user1 = users.sample
+
+    recipiant = Array.new(people, string)
+
+    while x < people
+      recipiant[x] = users.sample
+      x++
+    end
```

Comment:
Why make it so complicated? Take a look at the documentation for `Array#sample`: https://crystal-lang.org/api/0.24.1/Array.html#sample%28n%3AInt%2Crandom%3DRandom%3A%3ADEFAULT%29-instance-method